Tests/RegexTests/RegexDSLTests.swift

```diff
@@ -180,6 +180,31 @@ class RegexDSLTests: XCTestCase {
       }
     }
   }
+  
+  func testAssertions() throws {
+    try _testDSLCaptures(
+      ("aaaaab", "aaaaab"),
+      ("caaaaab", nil),
+      ("aaaaabc", nil),
+      captureType: Substring.self, ==)
+    {
+      Assertion.startOfLine
```

Comment:
Many of the built-in ones are more commonly called "anchors", which might be worth considering too.